src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -70,6 +71,8 @@
 
 #define MIN_SAFE_ACCEPT_QUEUE_SIZE 100
 
+int grpc_tcp_server_posix_expand_wildcard_addr = 0;
```

Comment:
We could define a server builder plugin that calls the vendor code, and include it in builds using that hardware.

The plugin (where it exists) would then set the channel arg, and that would control this expansion.